---
layout: post
title: "On the trail of evolutionary origin stories"
permalink: /evolutionaryorigin/
output: 
  html_document
  
---

It’s been a month since my end-of-first-year qualifying exam, and I’ve been meaning to write something about it. For the “quals” in my program (NYU Biology), each student picks two topics or areas of research, from which one is selected by the department. Based on the selected area of research, a committee of three professors is set up for the student, which then assigns a recent paper from that area. The student has to write a research proposal that builds on the findings of the assigned paper, and defend it to the committee. We defended over Zoom this year.  
  
  

My assigned paper, entitled Cnidarian Cell Type Diversity and Regulation Revealed by Whole-Organism Single-Cell RNA-Seq, was in many ways a modern take on the classic evolutionary biology paradigm of inferring the deep evolutionary past by comparing living organisms in the present. Evolutionary biologists often ask questions about origins. When and how did modern humans evolve? How did a structure as complex as the eye originate? When and how many times was Asian rice domesticated? How ancient are the genes used by mammals to fight viruses? Now, this paper’s findings on the origins of animal cell types made me think about how questions about origins and the tools used to answer them have evolved over the years.
  
    
    
Human fascination with the origins of natural phenomena isn’t new. Throughout human history, and until the 19th century, this preoccupation resulted in thousands of myths and theories. Darwin’s discovery of the theory of evolution by natural selection, and its implication of common evolutionary origin of very different organisms including humans represented the birth of a comparative approach to reconstructing biological origin stories.
  
    
    
In On the Origin of Species, Darwin famously and correctly concluded based on similarities in bone structures that the forelimbs of mammals and the wings of birds derive from a common origin. Such similarities had been noted before, notably by Pierre Belon in the 16th century, but never in an evolutionary context.
  
    
    
Reconstructions of this nature were prone to several problems. A handful of traits did not provide nearly enough statistical power to map evolutionary relationships between species with much confidence. To compound that, organisms could independently evolve the same trait without having shared a common origin, which Darwin recognized.
  
    
    
The early- to mid-twentieth century synthesis of genetics and evolution, soon accompanied by the advent of molecular biology, heralded a revolution in comparative studies that is still ongoing. The DNA sequences encoding the biological functions of organisms could now be compared between species. As sequencing technologies evolved, the number of “traits” used to infer evolutionary relationships between species increased hundredfold to more than a million-fold, as each nucleotide or letter in a DNA sequence is equivalent to a single trait. Advances in the understanding of the annual rate of acquiring changes or mutations in DNA allowed the approximate determination of how long ago different species diverged. Sequences of homologous genes across species could be compared to ask when certain genes first evolved.
  
    
    
Reconstructing the origins of genes, functions, or traits based on DNA sequences comes with its own set of limitations. Beyond a few hundred million years, DNA sequences can change substantially while retaining the same function, or remain recognizably similar while diverging in function. In multicellular organisms, the same gene can have very different roles depending on when, where, and how much of it is expressed, or in other words, gene regulation. Gene regulation is determined by non-coding DNA, which is subject to much more rapid change than genes, making sequence-based reconstruction difficult over longer evolutionary timeframes. In recent years, there has been remarkable progress in overcoming these limitations.
  
    
    
Functional conservation or differentiation of genes can sometimes be studied through experiments in cultured cells or model organisms, for instance by knocking out a gene and seeing its effects. In recent years, several studies have reconstructed ancestral genes based on the sequences of modern homologs, and actually expressed them in cells in order to assess their original function. A Nature study this year employed this approach to study the evolution of vertebrate hemoglobin, retracing the duplications and mutations that led to the origin of the tetrameric (having four protein units) modern hemoglobin from a monomeric (one protein unit) ancestral form more than 400 million years ago.
  
    
    
The development of single-cell RNA sequencing, which measures the expression of genes in each cell, has enabled arguably the most powerful use of the comparative approach to date. Genes rarely work in isolation, and usually constitute parts of pathways and networks of genes, many of which in turn help define specific cell types in multicellular organisms. By comparing gene expression in cells across organisms, it is now possible to infer the origins of specialized cell types and gene regulatory networks underlying specific functions.
  
     
     
My quals paper, published in Cell in 2018 performed single-cell RNA sequencing of every single cell from entire individuals of the sea anemone Nematostella vectensis. Sea anemones belong to a group called cnidarians, which diverged from a common ancestor that also led to bilaterians, which include everything from fruit flies and worms to frogs and humans. Therefore, comparisons between cnidarian and bilaterian species can yield inferences about their shared ancestor, and provide insight on whether specific traits evolved in bilaterians or had already existed before their divergence from the cnidarian-bilaterian ancestor more than 600 million years ago.
  
    
    
Many questions remain over the origins and functional diversification of cell types in animals, a group to which both cnidarians and bilaterians belong. Among the two, bilaterians appear to have significantly greater diversity and specialization of cell types, partly because cnidarians and their cells and body parts are visually less complex. This study, in sequencing every single cell from larval and adult sea anemones, was able to capture much greater diversity than previously thought.
  
     
     
Cells were classified based on the expression of genes with known function, or that are homologous to genes of known function in other species. For instance, myc is a gene that is associated with progenitor cells that can give rise to other cells across animals, and cells expressing myc and other similar markers were classified as putative progenitor cells.
  
    
    
This approach revealed, among other findings, a highly diverse repertoire of nerve cells or neurons. The authors then identified genes that were uniquely expressed in each Nematostella cell type and asked how old they were. If a gene that was only expressed in Nematostella neurons was found to be present only in cnidarians, it would be considered younger than a gene that was expressed in both cnidarians and bilaterians, in which case it must have arisen before the divergence of the two groups.
  
    
    
They found that while many neuron-specific genes are indeed ancient and existed before the divergence of cnidarians and bilaterians, a larger proportion of neuron-specific genes in Nematostella are only found in cnidarians. This suggests that neurons in cnidarians underwent functional diversification after splitting from the cnidarian-bilaterian ancestor. The ancestor therefore likely contained cells with neuronal or near-neuronal function, but it is hard to infer exactly how complex its neuronal gene repertoire was.
  
    
    
It’s still early days for the single-cell revolution, but it’s clear that its tools will be key in identifying mechanisms underlying key functional innovations, through allowing exploration of evolutionary change at the finer scales of cell types and networks of genes.
  
    
